File: app/src/main/java/expressive/fan/states/PausedState.java
package expressive.fan.states;

import expressive.fan.core.MainActivityController;

public class PausedState extends AbstractState {

    private MainActivityController controller;

    public PausedState(MainActivityController controller) {
        this.controller = controller;
        this.controller.setPlayButtonEnabled(true);
        this.controller.setPauseButtonEnabled(false);
        this.controller.setStopButtonEnabled(true);
    }

    @Override
    public void playClicked() {
        controller.play();
        controller.setNextState(new PlayingState(controller));
    }

    @Override
    public void stopClicked() {
        controller.stop();
        controller.setNextState(new StoppedState(controller));
    }

    @Override
    public void progressChanged(int progress) {
        int total = controller.getTotalAudioDuration();
        int newPos = total / 100 * progress;
        controller.seekTo(newPos);
    }
}

File: app/src/main/java/expressive/fan/states/PausedState.kt
package expressive.fan.states

import expressive.fan.core.MainActivityController

class PausedState(private val controller: MainActivityController) : AbstractState() {
    override fun playClicked() {
        controller.play()
        controller.setNextState(PlayingState(controller))
    }

    override fun stopClicked() {
        controller.stop()
        controller.setNextState(StoppedState(controller))
    }

    override fun progressChangedByUser(progress: Int) {
        controller.seekToNewPos(progress)
    }

    override fun progressChangedByScheduler() {
        controller.updateProgress()
    }

    init {
        controller.setPlayButtonEnabled(true)
        controller.setPauseButtonEnabled(false)
        controller.setStopButtonEnabled(true)
    }
}